code/frontend/MakeItSo/MakeItSo/MakeItSo.docc/Tutorials/Building a simple todo list UI.tutorial

```diff
@@ -0,0 +1,473 @@
+    @Steps {
+      @Step {
+        Define a property `completion` for the completion handler. This handler has one parameter for passing the `Reminder` to the calling view.
+        
+        Adjust the preview provider to make use of the completion handler. 
+        
+        @Code(name: "AddReminderView.swift", file: building-todo-ui-reminder-04-addreminderview-07.swift, previousFile: building-todo-ui-reminder-04-addreminderview-06) {
+          @Image(source: building-todo-ui-reminder-04-addreminderview-06.png, alt: "")
+        }
+      }
+      
+      @Step {
+        Add a new private function `commit`. Inside this function, call the completion handler, passing the `reminder` property.
+        
+        Call the `commit` function when the user taps on the _Add_ button. 
+        
+        @Code(name: "AddReminderView.swift", file: building-todo-ui-reminder-04-addreminderview-08.swift) {
+          @Image(source: building-todo-ui-reminder-04-addreminderview-06.png, alt: "")
+        }
+      }
+      
+      @Step {
+        Add a new private variable `dismiss` at the top of `AddReminderView`, and use the `@Environment` property wrapper to connect it to the `dismiss` action from the SwiftUI environment.
+        
+        @Code(name: "AddReminderView.swift", file: building-todo-ui-reminder-04-addreminderview-09.swift) {
+          @Image(source: building-todo-ui-reminder-04-addreminderview-06.png, alt: "")
+        }
+      }
+      
+      @Step {
+        Call `dismiss()` at the end of the `commit` function, after passing the new reminder to the calling view. This will dismiss the `Add Reminder` dialog after the data has been passed.
+        
+        > Note: This works because the `dismiss` environment value makes use of `callAsFunction`. To learn more about this, check out [How and when to use callAsFunction in Swift 5.2](https://www.donnywals.com/how-and-when-to-use-callasfunction-in-swift-5-2/) or [Callable values of user-defined nominal types – available from Swift 5.2](https://www.hackingwithswift.com/swift/5.2/callasfunction).
+        
+        @Code(name: "AddReminderView.swift", file: building-todo-ui-reminder-04-addreminderview-10.swift) {
+          @Image(source: building-todo-ui-reminder-04-addreminderview-06.png, alt: "")
+        }
+      }
+    }
+  }
+  
+  @Section(title: "Present the Add Reminder Dialog") {
+    @ContentAndMedia {
+      Open the dialog from the reminders lit.
+      
+      @Image(source: placeholder-image.png, alt: "")
+    }
+    
+    @Steps {
+      @Step {
+        Go back to the `ContentView.swift` file, and add a new boolean property `isAddReminderDialogPresented`.  
+        
+        @Code(name: "ContentView.swift", file: building-todo-ui-reminder-05-contentview-present-addreminder-01.swift, previousFile: building-todo-ui-reminder-03-contentview-04.swift)
+      }
+              
+      @Step {
+        Add a convenience function for toggling the state of `isAddReminderDialogPresented`
```

Comment:
Might be a dumb question (but maybe other people will ask the same): what's a convenience function?

Reply:
Definitely not a dumb question! I think (and Peter correct me if I'm wrong) that this function just serves to make the code read better when mutating `isAddReminderDialogPresented`? If so, maybe say "helper" function instead? Whenever I see "convenience" with Swift code, I think "convenience" initializers which isn't the case here.

optional: here and throughout tutorial
```suggestion
        Add a helper function for toggling the state of `isAddReminderDialogPresented`
```

Reply:
+1 to what @ncooke3 said. _Helper function_ is a much better word.

Reply:
Got it! Thanks for explaining.